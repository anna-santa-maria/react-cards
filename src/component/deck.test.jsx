import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Deck from './deck';
import GameModel from '../model/game_model';

configure({ adapter: new Adapter() });

describe('<Deck/>', () => {
    const props = {
        cards: GameModel.generate(4),
        onRemove: jest.fn(),
    };

    describe('render', () => {
        it('with default/required props', () => {
            const c = shallow(<Deck {...props} />);

            expect(c).toMatchSnapshot();
        });
    });
});
