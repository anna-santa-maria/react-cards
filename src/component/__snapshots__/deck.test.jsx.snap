// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Deck/> render with default/required props 1`] = `
<section
  className="deck"
>
  <Card
    key="0"
    onClick={[Function]}
    rank={14}
    type={1}
  />
  <Card
    key="1"
    onClick={[Function]}
    rank={14}
    type={2}
  />
  <Card
    key="2"
    onClick={[Function]}
    rank={14}
    type={4}
  />
  <Card
    key="3"
    onClick={[Function]}
    rank={14}
    type={8}
  />
</section>
`;
